### PR TITLE
change megadepth loss to MiDaS loss and change depth decoder

### PR DESCRIPTION
**Pull request by melisandeteng**
Please check that depth is correctly implemented from MiDaS paper

### CODE REVIEW

File: omnigan/blocks.py [hunk 1]
```diff
@@ -293,6 +293,42 @@ def __str__(self):
...
+    """#Depth decoder based on depth auxiliary task in DADA paper
+
+    """
+    def __init__(self, opts):
+        super().__init__()
+        self.relu = nn.ReLU(inplace = True)
+        self.enc4_1 = nn.Conv2d(2048, 512, kernel_size=1, stride=1, padding=0, bias=True)
+        self.enc4_2 = nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1, bias=True)
+        self.enc4_3 = nn.Conv2d(512, 128, kernel_size=1, stride=1, padding=0, bias=True)
+        self.enc4_1.weight.data.normal_(0, 0.01)
```
Comment (vict0rsch) [inline omnigan/blocks.py:306]: there's an automatic init function run in get_generator. Either you remove things here or add an if statement there so that things are not re-initialized

File: omnigan/blocks.py [hunk 2]
```diff
@@ -293,6 +293,42 @@ def __str__(self):
...
+        x4_enc = self.enc4_3(x4_enc)
+        
+        depth = torch.mean(x4_enc, dim=1, keepdim=True)  # DADA paper decoder
+        depth = F.interpolate(
+                    depth,
+                    size=(384,384), #size used in MiDaS inference
+                    mode="bicubic", #what MiDaS uses
+                    align_corners=False,
+                )
+        depth = F.interpolate(
```
Comment (vict0rsch) [inline omnigan/blocks.py:324]: have you looked at outputs? can you just try and profile that (use time()) to see the overhead of 2 interpolations?

File: omnigan/losses.py [hunk 3]
```diff
@@ -277,22 +277,45 @@ def __call__(self, prediction, target):
...
+        gmLoss = O #gradient matching term
+        for k in range(scale):
+            R_ = F.interpolate(R, scale_factor = 1/2**k)
+            Rx = F.conv2d(R_, self.sobelx, stride=1)
+            Ry = F.conv2d(R_, self.sobely, stride=1)
+            gmLoss += torch.sum(torch.abs(Rx) + torch.abs(Ry))
+        gmLoss = self.gmweight/ num_pix * gmLoss
+        #scale invariant MSE
+        simseLoss = 0.5/num_pix * torch.sum(torch.abs(R))
+        loss = simseLoss + gmLoss
```
Comment (vict0rsch) [inline omnigan/losses.py:317]: LGTM

**Review by vict0rsch — CHANGES_REQUESTED** [review]

**Comment by vict0rsch** [comment]
The only blocking point is the init to me otherwise good to run

File: omnigan/losses.py (hunk 3 above)
Comment (melisandeteng) [reply omnigan/losses.py:317]: I had to google that 😄

Comment (vict0rsch) [reply omnigan/losses.py:317]: Haha sorry 🙏 looks good to me (for future readers)

File: omnigan/blocks.py (hunk 1 above)
Comment (melisandeteng) [reply omnigan/blocks.py]: I changed that, please check that its ok, things are running

File: omnigan/blocks.py (hunk 2 above)
Comment (melisandeteng) [reply omnigan/blocks.py:321]: no not yet, not quite sure how bicubuc vs bicubuc + nearest compare, I'll do that later

File: omnigan/losses.py [hunk 4]
```diff
@@ -395,6 +418,7 @@ def get_losses(opts, verbose, device=None):
     # task losses
     # ? * add discriminator and gan loss to these task when no ground truth
     # ?   instead of noisy label
+
```
Comment (alexrey88) [inline omnigan/losses.py:421]: To be consistent with the rest of the code, maybe we should take the scaler out of the loss and put it in trainer.py? Like we did for TV Loss today.

**Review by vict0rsch — APPROVED** [review]
